README.md

```diff
@@ -9,7 +9,7 @@ The implementation comes in two formats:
 - Trie DB (`trie-db` crate) which can be combined with a backend database to provide
    a persistent trie structure whose contents can be modified and whose root hash
    is recalculated efficiently.
-- Trie Hash (`trie-hash` crate) which provides a closed-form function that accepts a
+- Trie Root (`trie-root` crate) which provides a closed-form function that accepts a
    sorted enumeration of keys and values (exactly the format provided by
```

Comment:
this doesn't seem to be up to date? `trie_root` sorts the iterator under the hood 
https://github.com/paritytech/trie/blob/e23681d2cd4f97bc1fe60e86f042bef7c0e93630/trie-root/src/lib.rs#L113-L116

Reply:
Oh, yes you are right (it kind of defeat the purpose of streaming the values, but in practice it would be need to be done by substrate, currently the primitives that run on ordered transaction (key being number of transaction) does a compact encoding of key (see encode_index method implementation of substrate) : that break ordering. If compact encoding was done in a way that preserve encoding, this sort would have been shippable).
Personally I think it would be good to remove this ordering from trie-root crate and put it explicitly in substrate. But for this PR I guess it is fine to remove the 'sorted' word and the note in parenthesis about the ordering. Also the line following the sort would need to be refactored, my hypothesis is that it is done like that for performance reason (extend all the nibbles at once), so I would rather not touch it, especially since I would prefer to use iter_build instead.
So finally I guess changing the doc is better here.

For info (just ran the test to be sure), compact encoding in substrate looks like:
https://github.com/paritytech/parity-scale-codec/blob/6fe7acdc806d82bf9ce3208e037c4d9f4be60a81/src/compact.rs#L720
if the compact scheme would retain ordering of key when serialized we could skip this key ordering step.

Reply:
Thanks for looking into this. Yeah it makes sense to just keep the readme up to date instead of changing the behavior here at least for now.